docs/security/access-control.md

```diff
@@ -0,0 +1,671 @@
+-H "Content-Type: application/json" -d '{
+   name: "role1",
+   properties: {"k1", "v1"}
+   securableObjects: [
+          {
+             fullName: "catalog1.schema1.table1",
+             type: "TABLE",
+             privileges: [
+                    {
+                         name: "SELECT_TABLE",
+                         condition: "ALLOW"
+                    }
+             ]    
+          }
+   ]
+}' http://localhost:8090/api/metalakes/test/roles
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+
+SecurableObject securableObject =
+    SecurableObjects.ofTable(
+        SecurableObjects.ofSchema(
+            SecurableObjects.ofCatalog("catalog1", Collections.emptyList()),
+            "schema1",
+            Collections.emptyList()),
+        "table1",
+        Lists.newArrayList(Privileges.SelectTable.allow()));
+      
+Role role =
+    client.createRole("role1", ImmutableMap.of("k1", "v1"), Lists.newArrayList(securableObject));
+```
+
+</TabItem>
+</Tabs>
+
+### Get a role
+
+You can get a role by its name.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X GET -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d http://localhost:8090/api/metalakes/test/roles/role1
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+Role role =
+    client.getRole("role1");
+```
+
+</TabItem>
+</Tabs>
+
+### Delete a role
+
+You can delete a role by its name.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X DELETE -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" http://localhost:8090/api/metalakes/test/roles/role1
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+boolean deleted =
+    client.deleteRole("role1");
+```
+
+</TabItem>
+</Tabs>
+
+## Permission Operation
+
+### Grant roles to a user
+
+You can grant specific roles to a user.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    roleNames: ["role1"]
+}'http://localhost:8090/api/metalakes/test/permissions/users/user1/grant
+
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+User user = client.grantRolesToUser(Lists.newList("role1"), "user1");
+```
+
+</TabItem>
+</Tabs>
+
+### Revoke roles from a user
+
+You can revoke specific roles from a user.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    roleNames: ["role1"]
+}'http://localhost:8090/api/metalakes/test/permissions/users/user1/revoke
+
```

Comment:
Remove this empty line in here and other places.

Reply:
OK.